<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
<ros2_control name="MagneckoVectornav" type="sensor">
    <hardware>
        <plugin>magnecko_vectornav/MagneckoVectornav</plugin>
        <param name="example_param_hw_start_duration_sec">0.0</param>
        <param name="example_param_hw_stop_duration_sec">3.0</param>
        <!-- <param name="vectornavConfigFilePath">${find magnecko_vectornav}/config/setup.yaml</param> -->
    </hardware>
    <sensor name="imu_sensor">
        <state_interface name="orientation.x" />
        <state_interface name="orientation.y" />
        <state_interface name="orientation.z" />
        <state_interface name="orientation.w" />
        <state_interface name="angular_velocity.x" />
        <state_interface name="angular_velocity.y" />
        <state_interface name="angular_velocity.z" />
        <state_interface name="linear_acceleration.x" />
        <state_interface name="linear_acceleration.y" />
        <state_interface name="linear_acceleration.z" />
        <state_interface name="time_stamp_sec" />
        <state_interface name="time_stamp_nano" />

        <!-- Declare setup parameters -->
        <!-- general parameters -->
        <param name="frame_id">vectornav</param>
        <param name="baud">115200</param>
        <param name="AsyncDataOutputType">0</param>
        <param name="AsyncDataOutputFrequency">20</param>
        <param name="syncInMode">3</param>
        <param name="syncInEdge">0</param>
        <param name="syncInSkipFactor">0</param>
        <param name="syncOutMode">0</param>
        <param name="syncOutPolarity">0</param>
        <param name="syncOutSkipFactor">0</param>
        <param name="syncOutPulseWidth_ns">100000000</param>
        <param name="serialCount">0</param>
        <param name="serialStatus">0</param>
        <param name="spiCount">0</param>
        <param name="spiStatus">0</param>
        <param name="serialChecksum">1</param>
        <param name="spiChecksum">0</param>
        <param name="errorMode">1</param>
        <!-- BO1 -->
        <param name="BO1.asyncMode">3</param>
        <param name="BO1.rateDivisor">40</param>
        <param name="BO1.commonField">0x7FFF</param>
        <param name="BO1.timeField">0x0000</param>
        <param name="BO1.imuField">0x0000</param>
        <param name="BO1.gpsField">0x0000</param>
        <param name="BO1.attitudeField">0x0000</param>
        <param name="BO1.insField">0x0000</param>
        <param name="BO1.gps2Field">0x0000</param>
        <!-- BO2 -->
        <param name="BO2.asyncMode">0</param>
        <param name="BO2.rateDivisor">0</param>
        <param name="BO2.commonField">0x0000</param>
        <param name="BO2.timeField">0x0000</param>
        <param name="BO2.imuField">0x0000</param>
        <param name="BO2.gpsField">0x0000</param>
        <param name="BO2.attitudeField">0x0000</param>
        <param name="BO2.insField">0x0000</param>
        <param name="BO2.gps2Field">0x0000</param>
        <!-- BO3 -->
        <param name="BO3.asyncMode">0</param>
        <param name="BO3.rateDivisor">0</param>
        <param name="BO3.commonField">0x0000</param>
        <param name="BO3.timeField">0x0000</param>
        <param name="BO3.imuField">0x0000</param>
        <param name="BO3.gpsField">0x0000</param>
        <param name="BO3.attitudeField">0x0000</param>
        <param name="BO3.insField">0x0000</param>
        <param name="BO3.gps2Field">0x0000</param>
    </sensor>
</ros2_control>
</robot>